packages/amplify-graphql-index-transformer/src/graphql-index-transformer.ts

```diff
@@ -69,7 +69,7 @@ export class IndexTransformer extends TransformerPluginBase {
   public after = (ctx: TransformerContextProvider): void => {
     if (!ctx.isProjectUsingDataStore()) return;
 
-    const overriddenResources = getResourceOverrides([this], ctx?.stackManager);
+    const overriddenResources = getResourceOverrides([this], ctx.filepaths.getBackendDirPath(), ctx?.stackManager);
```

Comment:
ctx is optional in the latter, can we rely on it in the former?

Reply:
I'm not really sure why we have optional chaining for `stackManager`. It doesn't look like it does anything in the case anyway because we call `isProjectUsingDataStore` on `ctx` on line 70.

Reply:
Looking at the original commit, I don't see any reasoning for adding the optional chaining here. And it would still have the same issue as today with calling `isProjectUsingDataStore`.

https://github.com/aws-amplify/amplify-category-api/commit/5e59d18e3341b00bfdcc2e5edabf016fed8eb986